### PR TITLE
Add additional sources fields to pensions form

### PR DESCRIPTION
**Pull request by jbalboni**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2911

This adds the additional sources widget to the disclosure pages as a custom field.

screen shot 2017-05-26 at 5 16 41 pm
screen shot 2017-05-26 at 5 16 46 pm
screen shot 2017-05-26 at 5 16 57 pm

### CODE REVIEW

File: src/js/common/schemaform/ArrayPage.jsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    const eligiblePageList = getActivePages(pageList, form.data);
+    const pageIndex = _.findIndex(item => item.pageKey === pageConfig.pageKey, eligiblePageList);
+    return { eligiblePageList, pageIndex };
+  }
+
+  goBack() {
+    const { eligiblePageList, pageIndex } = this.getEligiblePages();
+    // if we found the current page, go to previous one
+    // if not, go back to the beginning because they shouldn't be here
+    const page = pageIndex >= 0 ? pageIndex - 1 : 0;
```
Comment (cehsu) [inline src/js/common/schemaform/ArrayPage.jsx]: should this simply be > ?

File: src/js/pensions/components/AdditionalSourcesField.jsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                            formData={itemData}
+                            onChange={(value) => this.onItemChange(index, value, true)}
+                            onBlur={onBlur}
+                            registry={this.props.registry}
+                            disabled={disabled}
+                            readonly={readonly}/>
+                      </div>
+                      <div className="row small-collapse">
+                        <div className="small-6 left columns">
+                          <button className="float-left" onClick={() => this.handleUpdate(index)}>Update</button>
```
Comment (cehsu) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: Should this be "Save"?

**Review by cehsu — APPROVED** [review]
LGTM. Had a minor UI question, and will review again to see if can find any others, but looks good.

File: src/js/pensions/components/AdditionalSourcesField.jsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    this.handleRemove = this.handleRemove.bind(this);
+    this.scrollToRow = this.scrollToRow.bind(this);
+  }
+
+  componentWillReceiveProps(newProps) {
+    const items = newProps.formData || [];
+    if (items.length < this.state.editing.length) {
+      this.setState({ editing: this.state.editing.filter((item, index) => index < items.length) });
+    } else if (items.length > this.state.editing.length) {
+      this.setState({ editing: this.state.editing.concat(true) });
```
Comment (cehsu) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: Is the intended behavior to be able to update or to delete and create according to the original issue and the description provided: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2911?

Comment (cvalarida) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: :man_shrugging: Doesn't really look like it, but how annoying would it be to mistype something, save it, realize you mistyped and...well bugger. Can't do anything about it now. Can't update it, can't delete it and start over. Meh.

Comment (cehsu) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: @goldenmeanie thanks for sharing your thoughts on this issue that @cvalarida highlighted.

Comment (jbalboni) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: I asked Alex about this when I first started and there's not a way to go back and edit/remove a source with this design. The way to get around it is to set the amount to 0 and add another source. I think if that turns out to be confusing in user testing, we can revisit it (or if Alex wants to make a modification to the design).

@cehsu The GH issue and the mock are the same; the text listed at the bottom is for the editing state after you click "Add Another Source."

File: src/js/pensions/components/AdditionalSourcesField.jsx (hunk 2 above)
Comment (jbalboni) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: Yep, good catch.

File: src/js/pensions/components/AdditionalSourcesField.jsx (hunk 3 above)
Comment (cehsu) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: Thanks @jbalboni, sorry for my confusion @goldenmeanie.

Comment (goldenmeanie) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: All raised questions seem valid to me. Especially the label typo case. I am worried about over-tooling this part of the UI, but will definitely keep an eye out for this in testing.

Comment (goldenmeanie) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: We learned something new from stakeholders yesterday:
In the context of Pensions, we are more free to assume $0 barring input from applicants. I've made some updates to the prototype and will update in slack.

File: src/js/pensions/components/AdditionalSourcesField.jsx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    const { SchemaField } = registry.fields;
+
+    // if we have form data, use that, otherwise use an array with a single default object
+    const items = formData || [];
+
+    if (registry.formContext.reviewMode) {
+      return <ReviewSources sources={items}/>;
+    }
+
+    const hasItemsBeingEdited = this.state.editing.some(item => item);
```
Comment (jbalboni) [inline src/js/pensions/components/AdditionalSourcesField.jsx]: I'll change the name for that to make it more clear.

**Pull request merged**